README.md

```diff
@@ -4,19 +4,23 @@ Suppose there is a task to test code that relies on certain system functions. Id
 
 To simplify this process, this library allows you to replace imported module functions, including Win32 system functions, with a stub for testing purposes using GMock. Since it is an extension of the GMock library, you can use the same actions and matchers as you normally would with GMock.
 
+---
+
+Mocking of system functions in general can be a sign of poorly designed code. This means that the code is tightly coupled to the system and relies on a specific implementation with side effects, making it less flexible and difficult to test.
+
+But sometimes it may be impossible (or quite expensive) to rewrite existing code to decouple it from that system API before testing it. Also we must not forget that `wrappers over the system API must also be tested` (and it's usually not recommended to introduce an extra layer of abstraction solely for testing purposes). In such situations, we can use this GMock-lib extension to test that code (at least tests for system wrappers that many projects have).
+
```

Comment:
> It would be nice to add the keywords "Breaking Contract" for search engines to the documentation. This is a debated topic. With a “bad” architecture, due to the impossibility of making changes or legacy code, we have to make a "Breaking Contract".

I agree, that the words “sign of poorly designed code”  is not always an objective truth. Could you describe an idea about "Breaking Contract" (to put it into this part of the readme file)?

Reply:
I will try. Sometimes I write very poorly in human language, I need to practice more.